### ISSUE TITLE
Messages that aren't supposed to be accesible are still (kind of) visible

### ISSUE THREAD

**Issue by Solsticely** [syuilo/Misskey#8317]
## 💡 Summary

Private messages' content are substituted by parenthesis 
![image](https://user-images.githubusercontent.com/96320174/153989340-c9a988d6-7872-488f-8f33-dc0713baf022.png)

These messages are accessible through the user interface. I can spot a few just going to the "notes and responses" section of a profile and finding a publicly readable response to a private message.

## 🥰 Expected Behavior

Messages not meant to be seen by the user should not be visible to the user, or should atleast have a description along the lines of "This message is private/hidden/is part of a private discussion between other users/etc"

## 🤬 Actual Behavior

Parenthesies ( )w( )

## 📝 Steps to Reproduce

Here's a conversation: https://stop.voring.me/notes/8wprsu5kyb

A general guide to achieving this:

1. Find a profile
2. Go to their "Notes and Responses" section
3. Find a response to a private note (look for `()`s)
4. :tada: 

## 📌 Environment

Misskey version: 12.105.0
Your OS: Solus Budgie
Your browser: Tested on Firefox and Chromium

**Comment by Johann150** [comment]
I'm a bit confused as to what you think is the problem here, as I can see multiple issues here.

I would consider `()` a form of description that this content is unavailable. I don't think users should normally see this so it would be unnecessary effort to add a localisation phrase for this. If someone were to find the exact URL of a direct note they should not see, I think it would be more appropriate for the server to return a `404` error code (as to not give away that a note with this ID exists) and the client to display the respective error message. There should not be a special error message for this.
I tested this special case on Mastodon and it does not display the replied to messages at all.

The other issue is in my opinion that replies to direct notes should be required to also be direct notes. This is also the case on Pleroma (but not Mastodon).

**Comment by Johann150** [comment]
I discovered that this is also an i18n bug because the `private` localisation string was removed.

https://github.com/misskey-dev/misskey/blob/45462e4a5e8b94be110968afb2985bb145d7b6b3/packages/client/src/components/sub-note-content.vue#L4-L5 (note that also `deleted` does not exist as a localisation string)

https://github.com/misskey-dev/misskey/blob/b01a0325ba8b682da7fc44764d6abb9c320aa90e/packages/client/src/components/note.vue#L51

https://github.com/misskey-dev/misskey/blob/b01a0325ba8b682da7fc44764d6abb9c320aa90e/packages/client/src/components/note-detailed.vue#L65

**Comment by Solsticely** [comment]
I agree with the 404 idea! though that could cause some confusion if a user one opens a shared link to a private note. i'd be a happier soul w/ a custom error message (or maybe a 401 instead?), though that would imply that the note in question exists.

In my head, i was thinking of replacing the ()es with a "this message is private" thinking that the user was *supposed* to be able to see public replies to private posts. The title is,, misleading if not wrong. (or maybe the user is supposed to see them and i'm not getting this right ?)

**Comment by Johann150** [comment]
I think the user is not supposed to see them. The notes are private notes so they should never be displayed in public. Even the fact that these two people communicated should not be public since it can give away information. For example it could generate suspicions that two people are in a relationship, etc.

This also applies to reactions which should not be shown if viewed by someone that is not a recipient. See e.g. https://genau.qwertqwefsday.eu/notes/8xczvc4unc which is a direct note I sent to myself.

**Comment by Johann150** [comment]
## implementation suggestion
### backend
Remove the ability for any information about notes being fetched, if the note is not visible for the requesting user, as done by all other fediverse platforms to my knowledge.

In this case, the response of `note/show` should be modified:
- `replyId` and/or `renoteId` is `null`
- (if `detail: true` was requested) `reply` and/or `renote` is filled with a tombstone `{isHidden:true}`. It should not contain **any** information about the note. This could be implemented by e.g. changing [`Notes.hideNote`](https://github.com/misskey-dev/misskey/blob/d071d18dd7a394298d454d4ef6c5ca885164dd47/packages/backend/src/models/repositories/note.ts#L64).

All other endpoints which might indicate that a note exists should return an error (e.g. `noSuchNote`) if the note is not visible to the requesting user. This includes endpoints like `/api/note/create` (when responding to, renoting or quoting a note), `/api/note/translate` and more, mostly within `/api/note/`.

###  client

The client will have to be adjusted respectively to display a tombstone instead of the note; see fix suggestions in [above comment](https://github.com/misskey-dev/misskey/issues/8317#issuecomment-1040735041) but the UI design would probably have to be changed since there will be none of the usual information available so it can not just be implemented as a note with the text replaced. I think if the note itself is visible, it would be hard to hide that it is a response or quote renote, so a tombstone would be appropriate as discussed before.

---

@syuilo If this is okay I would like to go ahead with implementing this, otherwise your feedback is appreciated.

**Comment by Johann150** [comment]
Since my idea of a tombstone would not fit into the type `Packed<'Note'>` it would probably be a better idea if the tombstone is just `null`. Since `reply` and `renote` are already both `optional` and `nullable`, there can be a distinction between when they are `undefined` (no reply/renote) or `null` (reply/renote, but hidden). In that case the `isHidden` attribute can also be removed from `packedNoteSchema`.

**Comment by Johann150** [comment]
I am unsure if its a better idea to also raise an error for `notes/show` instead of using `null` as above. Raising an error would mean that you can not view a quote renote if the renoted note is not visible to you too, but I think it would be easier to implement.
